model/sql-query-model.js

```diff
@@ -47,10 +47,12 @@ exports.queryAllComments = async function(article_id)
     return allComments.rows;
 }
 
-exports.queryAllArticles = async function(topic)
+exports.queryAllArticles = async function(topic, sort_by, order)
 {
     const allArticles = await pool.query(
-        format("SELECT * FROM articles %s", (topic ? `WHERE topic='${topic}'` : "")));
+        format(`
+        SELECT * FROM articles %s
+        ORDER BY %s %s;`, (topic ? `WHERE topic='${topic}'` : ""), (sort_by ? sort_by : "date"), (order ? order : "ASC")));
```

Comment:
I, personally, like the conditional logic here but this is open to SQL injection and your variables need sanitising before being passed into the string litteral.